dashboard/app/controllers/pd/csf_certificate_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Pd::CsfCertificateController < ApplicationController
+  before_action :authenticate_user!
+
+  def generate_certificate
+    enrollment = params[:enrollment_code] ? Pd::Enrollment.find_by(code: params[:enrollment_code]) : nil
+
```

Comment:
This could also be loaded into `@enrollment` via CanCan's `load_resource` before action hook:

```ruby
load_resource :enrollment, class: 'Pd::Enrollment', find_by: :code, id_param: :enrollment_code
```